frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -300,33 +297,41 @@ function PlotlyFigure({
   }
 
   const reset = useCallback((): void => {
-    widgetMgr.setStringValue(element, "{}", { fromUi: true }, fragmentId)
-
-    // Reset the selection info within the plotly figure
-    setPlotlyFigure({
-      ...plotlyFigure,
-      data: plotlyFigure.data.map((trace: any) => {
-        return {
-          ...trace,
-          selectedpoints: [],
-        }
-      }),
-      layout: {
-        ...plotlyFigure.layout,
-        selections: [],
-      },
-    })
+    const dragmode = plotlyFigure.layout.dragmode
+    if ((dragmode && dragmode === "lasso") || dragmode == "select") {
+      widgetMgr.setStringValue(element, "{}", { fromUi: true }, fragmentId)
+    }
   }, [plotlyFigure, widgetMgr, element, fragmentId])
 
+  const onFormCleared = useCallback(() => {
+    const spec = applyTheming(element, initialFigure, theme)
+    // https://plotly.com/javascript/reference/layout/#layout-clickmode
+    // This allows single selections and shift click to add / remove selections
+    if (element.isSelectEnabled) {
+      spec.layout.clickmode = "event+select"
+      spec.layout.hovermode = "closest"
+    } else {
+      spec.layout.clickmode = initialFigure.layout.clickmode
+      spec.layout.hovermode = initialFigure.layout.hovermode
+    }
```

Comment:
What is the reason we would need to apply this?

Reply:
When you reset it, we need to make sure that you can still do click selections.